RecoCTPPS/PixelLocal/src/CTPPSPixelGainCalibrationDBService.cc

```diff
@@ -0,0 +1,39 @@
+// -*- C++ -*-
+//
+// Package:     RecoCTPPS/PixelLocal
+// Class  :     CTPPSPixelGainCalibrationDBService
+// 
+// Implementation:
+//     [Notes on implementation]
+//
+// Original Author:  Helio Nogima
+//         Created:  Thu, 23 Feb 2017 17:37:50 GMT
+//
+
+#include "RecoCTPPS/PixelLocal/interface/CTPPSPixelGainCalibrationDBService.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "CondFormats/CTPPSReadoutObjects/interface/CTPPSPixelGainCalibrations.h"
+#include "CondFormats/DataRecord/interface/CTPPSPixelGainCalibrationsRcd.h"
+
+CTPPSPixelGainCalibrationDBService::CTPPSPixelGainCalibrationDBService()
+{
+}
+
+CTPPSPixelGainCalibrationDBService::~CTPPSPixelGainCalibrationDBService()
+{
+}
+
+void CTPPSPixelGainCalibrationDBService::getDB(const edm::Event& iEvent, const edm::EventSetup& iSetup){
+
+ edm::eventsetup::EventSetupRecordKey recordKey(edm::eventsetup::EventSetupRecordKey::TypeTag::findType("CTPPSPixelGainCalibrationsRcd"));
+ if( recordKey.type() == edm::eventsetup::EventSetupRecordKey::TypeTag()) {
+  //record not found
+   edm::LogError(" CTPPSPixelGainCalibrationDBService")<<"Record CTPPSPixelGainCalibrationsRcd does not exist ";
```

Comment:
Yes, it should. I softened it with LogError to avoid too many crashes during test phase. Now the exception is back.